### PR TITLE
html5validator: init at 0.3.3

### PR DESCRIPTION
**Pull request by phunehehe**
###### Motivation for this change

Add new package

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by FRidh — CHANGES_REQUESTED** [review]
This is a cli tool only, please move it out of `python-modules` and `python-packages.nix`.

**Comment by phunehehe** [comment]
@FRidh done

File: pkgs/applications/misc/html5validator/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    rev = "v${version}";
+    sha256 = "130acqi0dsy3midg7hwslykzry6crr4ln6ia0f0avyywkz4bplsv";
+  };
+
+  meta = with stdenv.lib; {
+    description = "Command line tool that tests files for HTML5 validity";
+    license = licenses.mit;
+    maintainers = [maintainers.phunehehe];
+  };
+}
```
Comment (jonringer) [inline pkgs/applications/misc/html5validator/default.nix:22]: generally accepted layout in nixpkgs

```suggestion
{ buildPythonPackage, fetchFromGitHub, nose, openjdk, stdenv }:

buildPythonPackage rec {
  pname = "html5validator";
  version = "0.3.3";
  
    src = fetchFromGitHub {
    owner = "svenkreiss";
    repo = "html5validator";
    rev = "v${version}";
    sha256 = "130acqi0dsy3midg7hwslykzry6crr4ln6ia0f0avyywkz4bplsv";
  };

  propagatedBuildInputs = [openjdk];

  checkInputs = [nose];
  checkPhase = "PATH=$PATH:$out/bin nosetests";

  meta = with stdenv.lib; {
    description = "Command line tool that tests files for HTML5 validity";
    license = licenses.mit;
    maintainers = [maintainers.phunehehe];
  };
}
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -26814,4 +26814,8 @@ in
   vpsfree-client = callPackage ../tools/virtualization/vpsfree-client {};
 
   gpio-utils = callPackage ../os-specific/linux/kernel/gpio-utils.nix { };
+
+  html5validator = callPackage ../applications/misc/html5validator {
+    inherit (python36Packages) buildPythonPackage nose;
+  };
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:26820]: use python package scope
```suggestion
  html5validator = python36Packages.callPackage ../applications/misc/html5validator { };
```

Comment (jonringer) [inline pkgs/top-level/all-packages.nix:26820]: please attempt to sort this as well

File: pkgs/applications/misc/html5validator/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{ buildPythonPackage, fetchFromGitHub, nose, openjdk, stdenv }:
+buildPythonPackage rec {
+  pname = "html5validator";
+  version = "0.3.3";
+  propagatedBuildInputs = [openjdk];
+
+  checkInputs = [nose];
```
Comment (jonringer) [inline pkgs/applications/misc/html5validator/default.nix:7]:
```suggestion
  checkInputs = [ nose ];
```

File: pkgs/applications/misc/html5validator/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{ buildPythonPackage, fetchFromGitHub, nose, openjdk, stdenv }:
+buildPythonPackage rec {
+  pname = "html5validator";
+  version = "0.3.3";
+  propagatedBuildInputs = [openjdk];
```
Comment (jonringer) [inline pkgs/applications/misc/html5validator/default.nix:5]:
```suggestion
  propagatedBuildInputs = [ openjdk ];
```

File: pkgs/applications/misc/html5validator/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{ buildPythonPackage, fetchFromGitHub, nose, openjdk, stdenv }:
+buildPythonPackage rec {
```
Comment (jonringer) [inline pkgs/applications/misc/html5validator/default.nix:2]: it's not being exported in python-modules, so need to export library
```suggestion
buildPythonApplication rec {
```

**Comment by jonringer** [comment]
Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the commits.

This can be done with:
```
git reset HEAD~<number_of_commits - 1>
git add -- pkgs/
git commit --amend --no-edit
git push ... ... --force
```

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (phunehehe) [reply pkgs/top-level/all-packages.nix]: I moved this to the first plausible position

File: pkgs/applications/misc/html5validator/default.nix (hunk 1 above)
Comment (phunehehe) [reply pkgs/applications/misc/html5validator/default.nix]: applied

File: pkgs/applications/misc/html5validator/default.nix [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    sha256 = "130acqi0dsy3midg7hwslykzry6crr4ln6ia0f0avyywkz4bplsv";
+  };
+
+  propagatedBuildInputs = [ openjdk ];
+
+  checkInputs = [ nose ];
+  checkPhase = "PATH=$PATH:$out/bin nosetests";
+
+  meta = with stdenv.lib; {
+    description = "Command line tool that tests files for HTML5 validity";
```
Comment (jonringer) [inline pkgs/applications/misc/html5validator/default.nix:19]: please add homepage, this can just point to the github repo

Comment (phunehehe) [reply pkgs/applications/misc/html5validator/default.nix:20]: done

File: pkgs/applications/misc/html5validator/default.nix [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{ buildPythonApplication, fetchFromGitHub, nose, openjdk, stdenv }:
```
Comment (jonringer) [inline pkgs/applications/misc/html5validator/default.nix:1]:
```suggestion
{ buildPythonApplication, fetchFromGitHub, nose, openjdk, stdenv }:

```

Comment (jonringer) [inline pkgs/applications/misc/html5validator/default.nix:1]:
```suggestion
{ buildPythonApplication, fetchFromGitHub, nose, openjdk, lib }:
```

File: pkgs/applications/misc/html5validator/default.nix [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    rev = "v${version}";
+    sha256 = "130acqi0dsy3midg7hwslykzry6crr4ln6ia0f0avyywkz4bplsv";
+  };
+
+  propagatedBuildInputs = [ openjdk ];
+
+  checkInputs = [ nose ];
+  checkPhase = "PATH=$PATH:$out/bin nosetests";
+
+  meta = with stdenv.lib; {
```
Comment (jonringer) [inline pkgs/applications/misc/html5validator/default.nix:18]: avoid unnecessary stdenv
```suggestion
  meta = with lib; {
```

**Review by jonringer — COMMENTED** [review]
some nitpicks that I overlooked

**Comment by phunehehe** [comment]
@jonringer suggestions applied

**Review by infinisil — APPROVED** [review]
All review points have been addressed, this looks good!

**Pull request merged**